src/common/ui/code.vue

```diff
@@ -66,6 +66,9 @@ import 'codemirror/addon/search/match-highlighter';
 import 'codemirror/addon/search/searchcursor';
 import 'codemirror/addon/selection/active-line';
 import 'codemirror/keymap/sublime';
+import 'codemirror/addon/hint/show-hint.css';
+import 'codemirror/addon/hint/show-hint';
+import 'codemirror/addon/hint/anyword-hint';
```

Comment:
It would make sense to include `codemirror/addon/hint/javascript-hint` too.

Reply:
We could add it. But in my experience, the bundled `javascript-hint` is somewhat limited and could be confusing.

E.g., it has no troubles in providing proper selection while completing on  browser's globals, e.g., providing all function when completing `document.`, but it doesn't infer much more:

```javascript
  var el = document.querySelector('#some-id');
  // cannot auto complete, javascript-hint cannot tell el if an HTMLElement.
  el.
```

Reply:
One can try out `javascript-hint` in CodeMirror's online demo to see if it's worth adding:
  https://codemirror.net/demo/complete.html

Reply:
It provides completion for the standard JS methods and keywords so it makes sense and I don't understand the reluctance.

Reply:
I just feel `javascript-hint` is a bit limited and could be counter-intuitive that it works sometimes but not other times from end users perspective.

E.g., it can complete `querySelector` when the code is `document.q` (`document` is a global), but it cannot complete when the code is `someElt.q` (`someElt` is an HTML element object).

Reply:
If we include `javascript-hint`, what'd be the logic? One proposal is to first use `javascript-hint`, if it returns nothing, fallback to `anyword-hint`.

Reply:
Ah, I see. Yes, js+any sounds like it could solve the problem.